spectator-reg-metrics3/src/main/java/com/netflix/spectator/metrics3/MetricsRegistry.java

```diff
@@ -57,4 +54,42 @@ private String toMetricName(Id id) {
     final String name = toMetricName(id);
     return new MetricsTimer(clock(), id, impl.timer(name));
   }
+
+  @Override public <T extends Number> T gauge(Id id, T number) {
+    return gauge(id, number, (ToDoubleFunction<T>) value -> number.doubleValue());
+  }
+
+  @Override public <T> T gauge(Id id, T obj, ToDoubleFunction<T> f) {
+    final String name = toMetricName(id);
+
+    Gauge aggrGauge = this.impl.getGauges().get(name);
+
+    // Wasn't registered already
+    if (aggrGauge == null) {
+      aggrGauge = new MetricsGaugeAggr();
+      this.impl.register(name, aggrGauge);
```

Comment:
Hmm thats true, didn't thought about that, i'm not an expert in concurrency, could you suggest a fix for that? Should i sincronize the get and register part? I don't know if it is the better alternative due to synchronisation overhead, but since this method will not be called so many times, maybe works. Anyway, i'm waiting your feedback